src/server/pkg/metrics/metrics.go

```diff
@@ -1,64 +1,125 @@
 package metrics
 
 import (
+	"fmt"
 	"time"
 
+	"github.com/pachyderm/pachyderm/src/client/pkg/config"
 	"github.com/pachyderm/pachyderm/src/client/pkg/uuid"
 	"github.com/pachyderm/pachyderm/src/client/version"
 	db "github.com/pachyderm/pachyderm/src/server/pfs/db"
 
 	"github.com/dancannon/gorethink"
+	"github.com/segmentio/analytics-go"
 	"go.pedge.io/lion/proto"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc/metadata"
 	kube "k8s.io/kubernetes/pkg/client/unversioned"
 )
 
-func dbMetrics(dbClient *gorethink.Session, pfsDbName string, ppsDbName string, metrics *Metrics) {
+var globalReporter *Reporter
```

Comment:
Good question. A few reasons:

1) I need a way to tell if metrics are enabled

We have the appEnv.Metrics value in pachd's main function, but we need to check if metrics are enabled when the tracking methods are called from each API function.

2) We want to re-use the segment client (since this is what does the batching for us) 

What I can do is break those out into two global objects so the rest of the 'reporter' isn't global. That's easy enough (and the client itself used to be a global). While thats technically more global objects, its much more concise about what we need to be global, which I think is where some of your concern is coming from.